hack/istio/multicluster/setup-external-controlplane.sh

```diff
@@ -152,7 +153,7 @@ spec:
         VALIDATION_WEBHOOK_CONFIG_NAME: "istio-validator-external-istiod-external-istiod"
         EXTERNAL_ISTIOD: "true"
         LOCAL_CLUSTER_SECRET_WATCHER: "true"
-        CLUSTER_ID: ${REMOTE_CLUSTER_NAME}
+        CLUSTER_ID: ${EXTERNAL_CLUSTER_NAME}
```

Comment:
@nrfox I think this was actually the reason for my test failure in CI. I think the new code actually failed a test that was passing before, but probably should not have been. Tell me if you agree. I also updated `mesh_test.go` to actually look for the correct external controlplane name.

Reply:
No this should be: `REMOTE_CLUSTER_NAME`. Look at step 3 in the [external controlplane deployment instructions](https://istio.io/latest/docs/setup/install/external-controlplane/#set-up-the-control-plane-in-the-external-cluster).

The `CLUSTER_ID` gets set to the remote cluster. This is so that the istiod in the external controlplane cluster can act like a primary inside of the remote cluster.

In the external controlplane setup, the cluster where the controlplane is running is not included in `controlPlane.ManagedClusters`.